lib/tests/test_userdb.c

```diff
@@ -23,34 +23,43 @@
  */
 
 #include "userdb.h"
+#include <criterion/criterion.h>
 
-int main(void)
+Test(user_db, resolve_user_root)
 {
   gint uid;
-  gint gid;
 
-  if (!resolve_user("root", &uid))
-    return 1;
+  cr_assert(resolve_user("root", &uid));
+}
 
-  if (!resolve_user("0", &uid) || uid != 0)
-    return 1;
+Test(user_db, resolve_user_0)
+{
+  gint uid;
 
-  if (!resolve_user("-1", &uid) || uid != -1)
-    return 1;
+  cr_assert(resolve_user("0", &uid));
+  cr_assert_eq(uid, 0);
+}
+
+Test(user_db, resolve_group)
+{
+  gint gid;
 
-  if (resolve_group("nincsily", &gid))
```

Comment:
I think this was meant as `resolve_user` instead of group, because later it has a group check.
This is not part of the unit test conversation, but it would be nice if you could add a test for the `resolve_user` also.